modules/core/petra/petra-sql-dsl-spi/src/test/java/com/liferay/petra/sql/dsl/spi/SQLDSLTest.java

```diff
@@ -977,6 +977,52 @@ public void testPredicateNot() {
 					MainExampleTable.INSTANCE.flagColumn.eq(0))));
 	}
 
+	@Test
+	public void testPredicateNotOf() {
+		Predicate leftPredicate = MainExampleTable.INSTANCE.nameColumn.eq(
+			"one");
+		Predicate rightPredicate = MainExampleTable.INSTANCE.nameColumn.eq(
+			"two");
+
+		DefaultPredicate defaultPredicate = new DefaultPredicate(
+			leftPredicate, Operand.OR, rightPredicate);
+
+		Predicate ltePredicate =
+			MainExampleTable.INSTANCE.mainExampleIdColumn.lte(3L);
+
+		Predicate notLtePredicate = Predicate.not(ltePredicate);
+
+		String informalQuery =
+			"MainExample.mainExampleId <= ? not (MainExample.mainExampleId " +
+				"<= ?) and MainExample.name = ? or MainExample.name = ?";
+
+		Assert.assertEquals(
+			informalQuery,
+			String.valueOf(notLtePredicate.and(defaultPredicate)));
```

Comment:
This assertion exposes that `Predicate.not` does not work correctly. It makes an informal query.